### PR TITLE
feat(windows): rename windows agent to coder.exe

### PR DESCRIPTION
**Pull request by ghuntley**
Smallest possible change to improve https://github.com/coder/coder/issues/5050

### CODE REVIEW

File: provisionersdk/scripts/bootstrap_windows.ps1 [hunk 1]
```diff
@@ -15,7 +15,7 @@ while ($true) {
 		# https://github.com/microsoft/vscode-remote-release/issues/5699
 		$BINARY_URL="${ACCESS_URL}/bin/coder-windows-${ARCH}.exe"
 		Write-Output "Fetching coder agent from ${BINARY_URL}"
-		Invoke-WebRequest -Uri "${BINARY_URL}" -OutFile $env:TEMP\sshd.exe
+		Invoke-WebRequest -Uri "${BINARY_URL}" -OutFile $env:TEMP\coderd.exe
```
Comment (ghuntley) [inline provisionersdk/scripts/bootstrap_windows.ps1:18]: **question:** shall we call it `coder.exe` or `coderd.exe`?

**Review by kylecarbs — COMMENTED** [review]
It actually needs to be called sshd for VS Code Remote to work 🤦‍♂️

**Comment by kylecarbs** [comment]
See: https://github.com/microsoft/vscode-remote-release/issues/5699

It's very sad that we need to call it `sshd`

**Comment by mafredri** [comment]
Perhaps we could create a [hardlink using `fsutil`](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/fsutil-hardlink) so that we have both `coder.exe` and `sshd.exe`?

I'm not sure if Windows could be running on top of a fringe filesystem that results in a hard-linking error, but it could be a "best effort" attempt to have both commands available.

**Pull request closed without merging**